src/core/server/rendering/rendering_service.tsx

```diff
@@ -103,6 +103,9 @@ export class RenderingService {
       buildNum,
     });
 
+    const publicPlugins = [...uiPlugins.public].filter(
+      ([, plugin]) => includeUserSettings || plugin.enableForAnonymousPages
```

Comment:
(not necessarily in this PR, but) We really need to rename this `includeUserSettings` option that no longer reflect what this option does or means.

Reply:
I started to refactor this but I ran into unexplained TS errors. 

Agree it should be inverted and renamed to `isAnonymousPage` or something, will leave it for another day.